src/uct/cuda/cuda_copy/cuda_copy_ep.c

```diff
@@ -41,55 +43,107 @@ UCS_CLASS_DEFINE_DELETE_FUNC(uct_cuda_copy_ep_t, uct_ep_t);
     ucs_trace_data("%s [ptr %p len %zu] to 0x%" PRIx64, _name, (_iov)->buffer, \
                    (_iov)->length, (_remote_addr))
 
-#define UCT_CUDA_COPY_CHECK_AND_CREATE_STREAM(_iface, _id) \
-    if ((_iface)->stream[_id] == 0) { \
-        ucs_status_t __status; \
-        __status = UCT_CUDAR_CALL_LOG_ERR(cudaStreamCreateWithFlags, \
-                                          &(_iface)->stream[_id], \
-                                          cudaStreamNonBlocking); \
-        if (UCS_OK != __status) { \
-            return UCS_ERR_IO_ERROR; \
-        } \
+static UCS_F_ALWAYS_INLINE cudaStream_t *
+uct_cuda_copy_get_stream(uct_cuda_copy_iface_t *iface,
+                         ucs_memory_type_t src_type,
+                         ucs_memory_type_t dst_type)
+{
+    cudaStream_t *stream;
+    ucs_status_t status;
+
+    if ((src_type == UCS_MEMORY_TYPE_LAST) &&
+        (dst_type == UCS_MEMORY_TYPE_LAST)) {
+        stream = &iface->short_stream;
```

Comment:
why need special case and not use [LAST][LAST] entry?

Reply:
Removed the special handling and directly using short stream in short operations now.